### ISSUE TITLE
No Error or action on register-psrepository

### ISSUE THREAD

**Issue by BatmanAMA** [OneGet/oneget#390]
## Action
Register-PSRepository -Default

## Expected Responses

* PSGallery is registered
* An Error Occurs

## Actual Response

Nothing occurs

## Additional Information

I set $DebugPreference to true and I see it's trying to ping a different url (https://go.microsoft.com/fwlink/?LinkID=397631&clcid=0x409) entirely. This fails due to firewall restrictions and the command returns $Null instead of throwing an error.

```powershell
PS C:\Users\me> Register-PSRepository -Default -Verbose
DEBUG: 00:00:00.0000004 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0000569 Verbose: True
DEBUG: 00:00:00.0000891 Name: PSGallery
DEBUG: 00:00:00.0052884 INVOKING PowerShell Fn Get-DynamicOptions with args Provider that has length 1
DEBUG: 00:00:00.0097704 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0150141 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0300794 Calling New() : MethodName = 'GetDynamicOptions'
DEBUG: 00:00:00.0301241 Verbose: True
DEBUG: 00:00:00.0301549 Name: PSGallery
DEBUG: 00:00:00.0343211 INVOKING PowerShell Fn Get-DynamicOptions with args Source that has length 1
DEBUG: 00:00:00.0393242 In PowerShellGet Provider - 'Get-DynamicOptions'.
DEBUG: 00:00:00.0471766 Done calling powershell «Get-DynamicOptions» «PSModule»
DEBUG: 00:00:00.0960446 Calling New() : MethodName = 'ResolvePackageSources'
DEBUG: 00:00:00.0983069 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.1022021 ProviderName: PowerShellGet
DEBUG: 00:00:00.1055618 Verbose: True
DEBUG: 00:00:00.1065135 Name: PSGallery
DEBUG: 00:00:00.1162346 INVOKING PowerShell Fn Resolve-PackageSource with args  that has length 0
DEBUG: 00:00:00.1222437 In PowerShellGet Provider - 'Resolve-PackageSource'.
DEBUG: 00:00:00.1246643 PowerShell Script 'PSModule' Function 'Resolve-PackageSource' returns null.
DEBUG: 00:00:00.1289232 Done calling powershell «Resolve-PackageSource» «PSModule»
VERBOSE: Performing the operation "Register Module Repository." on target "Module Repository 'PSGallery' () in provider
 'PowerShellGet'.".
DEBUG: 00:00:00.1510380 Calling New() : MethodName = 'AddPackageSource'
DEBUG: 00:00:00.1531541 MessageResolver: Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.1568870 ProviderName: PowerShellGet
DEBUG: 00:00:00.1594885 Verbose: True
DEBUG: 00:00:00.1616982 Name: PSGallery
DEBUG: 00:00:00.1687573 INVOKING PowerShell Fn Add-PackageSource with args PSGallery, , False that has length 3
DEBUG: 00:00:00.1825840 In PowerShellGet Provider - 'Add-PackageSource'.
DEBUG: 00:00:00.1843477 OPTION: MessageResolver => Microsoft.PowerShell.PackageManagement.Cmdlets.GetMessageString
DEBUG: 00:00:00.1866136 OPTION: ProviderName => PowerShellGet
DEBUG: 00:00:00.1904075 OPTION: Verbose => True
DEBUG: 00:00:00.1928959 OPTION: Name => PSGallery
DEBUG: 00:00:21.2279480 Ping-Endpoint: location=https://go.microsoft.com/fwlink/?LinkID=397631&clcid=0x409,
statuscode=, resolvedLocation=
DEBUG: 00:00:42.2428586 Ping-Endpoint: location=https://go.microsoft.com/fwlink/?LinkID=622995&clcid=0x409,
statuscode=, resolvedLocation=
DEBUG: 00:00:42.2496064 PowerShell Script 'PSModule' Function 'Add-PackageSource' returns null.
DEBUG: 00:00:42.2555906 Done calling powershell «Add-PackageSource» «PSModule»
```

## My feedback

When this ping-endpoint fails, it should absolutely throw an error. I also would recommend pinging what that resolves to (https://www.powershellgallery.com/api/v2/items/psscript) instead so that firewall rules allowing https://www.powershellgallery.com/ work without needing to know this other URL

**Comment by alerickson** [comment]
Could you move this issue to the [PowerShellGet](https://github.com/PowerShell/PowerShellGet/issues) repository since register-psrepository is a PowerShellGet cmdlet?

**Comment by BatmanAMA** [comment]
opened [PowerShellGet#350](https://github.com/PowerShell/PowerShellGet/issues/350) in reference for this. Thanks for pointing me the right direction.

**Comment by RamonBertrand** [comment]
I know this is the wrong location to post this.

Ran into this issue and was unable to find a solution. The only link that properly represents the problem I was experiencing is this bug report. So, to whom it might concern:

I was able to reset my default repository by copying the following file from a machine where the error did not occur. C:\Users\<username>\AppData\Local\Microsoft\Windows\PowerShell\PowerShellGet\PSRepositories.xml

It seems to me like this is the only way to reset PSGallery when the default registration is missing. The -Default switch did not do anything.
